bpf/lib/encap.h

```diff
@@ -233,11 +235,32 @@ encap_and_redirect_netdev(struct __ctx_buff *ctx, struct tunnel_key *k,
 }
 #endif /* TUNNEL_MODE */
 
+static __always_inline __be16 tunnel_gen_src_port_v4(void)
```

Comment:
Are you planning to implement this as a follow-on?

Does this function generate a source port using the flow hash like that?
https://github.com/torvalds/linux/blob/0d85b27b0cc6b5cf54567c5ad913a247a71583ce/include/net/udp.h#L206

Reply:
Yep, just a detail that can follow afterwards once all the core infrastructure is sorted.

Reply:
This is for RSS steering on remote nodes, right? Perhaps rss_gen_src6() could be reused.

Reply:
> This is for RSS steering on remote nodes, right? 

yeah, the Geneve RFC says for instance:

> This source port SHOULD be the same for all packets belonging to a single encapsulated flow to prevent reordering due to the use of different paths. To encourage an even distribution of flows across multiple links, the source port SHOULD be calculated using a hash of the encapsulated packet headers using, for example, a traditional 5-tuple.

So I'll definitely have that follow-on PR in before `v1.14` ships.
 

>Perhaps rss_gen_src{4,6}() could be reused given we also hash in info from inner pkt.

Pretty much, yes (except that it needs to spit out a `__be16`). We also have `hash_from_tuple_v4()` :).

Reply:
Playing around in https://github.com/cilium/cilium/pull/26177.